lightning-transaction-sync/src/error.rs

```diff
@@ -0,0 +1,63 @@
+use std::fmt;
+
+#[derive(Debug)]
+/// An error that possibly needs to be handled by the user.
+pub enum TxSyncError {
+	/// A transaction sync failed and needs to be retried eventually.
+	Failed,
```

Comment:
Do we need an enum still? Could possibly have a variant for exposing the client error and another for the invalid merkle block. Not sure how much information you want to provide back.

Reply:
Yes, I'd like to keep it around for the time being, as more error variants may be introduced with the Electrum version.
As for exposing underlying client errors, unfortunately the `esplora_client::Error` type is [kind of a mess](https://docs.rs/esplora-client/0.3.0/esplora_client/enum.Error.html), so I'd want to make sure that I only expose some specific error conditions.

All in all I'd like to keep this around for now and expose more (unified) error types that subsume both Electrum and Esplora variants in a follow-up once both versions have landed.